src/fat_beagle.cpp

```diff
@@ -458,28 +475,36 @@ std::vector<double> FatBeagle::SubstitutionModelGradient(
   param_vector.segment(subst_map.at(SubstitutionModel::rates_key_).first,
                        subst_map.at(SubstitutionModel::rates_key_).second) =
       phylo_model_->GetSubstitutionModel()->GetRates();
-  // #324: make delta part of a gradient request
-  double delta = 1.e-6;
-  std::vector<double> frequencies_grad = SubstitutionModelGradientFiniteDifference(
-      f, fat_beagle, tree, subst_model, SubstitutionModel::frequencies_key_,
-      param_vector, delta, StickBreakingTransform());
-
-  std::vector<double> gradient;
-  // Rates in the GTR model are constrained to sum to 1
-  if (subst_model->GetRates().size() == 6) {
-    gradient = SubstitutionModelGradientFiniteDifference(
-        f, fat_beagle, tree, subst_model, SubstitutionModel::rates_key_, param_vector,
-        delta, StickBreakingTransform());
-  } else {
-    gradient = SubstitutionModelGradientFiniteDifference(
-        f, fat_beagle, tree, subst_model, SubstitutionModel::rates_key_, param_vector,
-        delta);
+  // Compute gradients with delta.
+  double delta = PhyloFlags::GetFlagValueIfSet(
+      flags, PhyloFlagGradientOptions::set_gradient_delta_, 1.e-6);
+  // Compute frequency gradients.
+  std::vector<double> freqs_grad;
+  if (do_freqs_grad) {
+    freqs_grad = SubstitutionModelGradientFiniteDifference(
+        f, fat_beagle, tree, subst_model, SubstitutionModel::frequencies_key_,
+        param_vector, delta, StickBreakingTransform());
   }
-  gradient.insert(gradient.end(), frequencies_grad.begin(), frequencies_grad.end());
-  return gradient;
+  // Compute rate gradients
+  std::vector<double> rates_grad;
+  if (do_rates_grad) {
+    // Rates in the GTR model are constrained to sum to 1
+    if (subst_model->GetRates().size() == 6) {
+      rates_grad = SubstitutionModelGradientFiniteDifference(
+          f, fat_beagle, tree, subst_model, SubstitutionModel::rates_key_, param_vector,
+          delta, StickBreakingTransform());
+    } else {
+      rates_grad = SubstitutionModelGradientFiniteDifference(
+          f, fat_beagle, tree, subst_model, SubstitutionModel::rates_key_, param_vector,
+          delta);
+    }
+  }
+  // Compile results.
+  return std::make_pair(rates_grad, freqs_grad);
 }
 
-PhyloGradient FatBeagle::Gradient(const UnrootedTree &in_tree) const {
+PhyloGradient FatBeagle::Gradient(const UnrootedTree &in_tree,
```

Comment:
This method does not use the PhyloFlags

Reply:
Added flags to routines to the UnrootedTree function.